File: leetcode/AIcoding/3.DeepSeek-v3/python_solutions/2272-substring-with-largest-variance.py
class Solution:
    def largestVariance(self, s: str) -> int:
        res = 0
        unique_chars = set(s)

        for a in unique_chars:
            for b in unique_chars:
                if a == b:
                    continue

                max_var = 0
                has_b = False
                first_b = False

                current = 0

                for ch in s:
                    if ch == a:
                        current += 1
                    elif ch == b:
                        has_b = True
                        if first_b and current >= 0:
                            first_b = False
                        else:
                            current -= 1
                            if current < 0:
                                first_b = True
                                current = -1

                    if has_b:
                        max_var = max(max_var, current)

                res = max(res, max_var)

        return res

File: leetcode/AIcoding/3.DeepSeek-v3/cpp_solutions/2272-substring-with-largest-variance.cpp
class Solution {
public:
    int largestVariance(string s) {
        int res = 0;
        unordered_set<char> unique_chars(s.begin(), s.end());

        for (char a : unique_chars) {
            for (char b : unique_chars) {
                if (a == b) continue;

                int var = 0;
                bool has_b = false, first_b = false;

                for (char ch : s) {
                    if (ch == a) var++;
                    else if (ch == b) {
                        has_b = true;
                        if (first_b && var >= 0) {
                            first_b = false;
                        } else if (--var < 0) {
                            first_b = true;
                            var = -1;
                        }
                    }
                    if (has_b) res = max(res, var);
                }
            }
        }

        return res;
    }
};